### PR TITLE
feat: use `@rnx-kit/react-native-host`

### PR DESCRIPTION
**Pull request by tido64**
### Description

React Native hosting bits have been extracted to a separate package.

### Platforms affected

- [ ] Android
- [x] iOS
- [x] macOS
- [ ] Windows

### Test plan

CI should build. Test apps on affected platforms should successfully boot.

#### iOS

```
cd example
pod install --project-directory=ios
yarn ios
```

#### macOS

```
cd example
pod install --project-directory=macos
yarn macos
```

#### iOS + New Architecture

```
npm run set-react-version 0.72
cd example
RCT_NEW_ARCH_ENABLED=1 pod install --project-directory=ios
yarn ios
```

### CODE REVIEW

File: ios/ReactTestApp/ContentView.swift [hunk 1]
```diff
@@ -53,7 +53,7 @@ final class ContentViewController: UITableViewController {
 
     override public func motionEnded(_: UIEvent.EventSubtype, with event: UIEvent?) {
         guard event?.subtype == .motionShake,
-              let bridge = reactInstance.bridge,
+              let bridge = reactInstance.host?.bridge,
```
Comment (Saadnajmi) [inline ios/ReactTestApp/ContentView.swift:56]: Any reason not to follow the same structure as lines 196-197?
```
guard let host = reactInstance.host,
      let bridge = host.bridge,
```

**Review by Saadnajmi — APPROVED** [review]

File: ios/ReactTestApp/ContentView.swift (hunk 1 above)
Comment (tido64) [reply ios/ReactTestApp/ContentView.swift:56]: I normally don't create a separate variable if it's not used outside of the block.

**Review by kelset — APPROVED** [review]
all works 👍

a couple of small hiccups but nothing strictly related to this PR, so we can proceed with it

**Comment by tido64** [comment]
> all works 👍
> 
> a couple of small hiccups but nothing strictly related to this PR, so we can proceed with it

Thanks ❤️

**Pull request merged**